src/Crawler.php

```diff
@@ -457,11 +457,6 @@ protected function createRobotsTxt(UriInterface $uri): RobotsTxt
     protected function getCrawlRequests(): Generator
     {
         while ($crawlUrl = $this->crawlQueue->getFirstPendingUrl()) {
-            if (! $this->crawlProfile->shouldCrawl($crawlUrl->url)) {
-                $this->crawlQueue->markAsProcessed($crawlUrl);
-                continue;
```

Comment:
Shouldn't we still have this `continue` though? Because with this change all logic in this while loop will get executed if the profile determines that it shouldn't crawl the url.